### ISSUE TITLE
Generalize and consolidate build system for porting to a new platform

### ISSUE THREAD

**Issue by jwolff-ncar** [NOAA-EMC/regional_workflow#176]
People should not have to modify dozens of different files in different places to port to their system, nor should they have to do so to add support for a different compiler or other type of different configuration. Setting up module files should also not be a requirement.

- Config file for different platforms

- Centralized location 

- Build on what global did (Dom, Laurie) 
   - For global: One configuration for libraries (incl pre/post), CIME has a config file (for model)

- Clean script to run before recompile

**Comment by jwolff-ncar** [comment]
@mkavulich Can you update this issue with information from the release build focus team? Should we close this issue or link it to other open issues that are related to keep tabs on progress?

**Comment by mkavulich** [comment]
This will be achieved once the top-level cmake build is in place. This is waiting on UFS_UTILS PR#118 (https://github.com/NOAA-EMC/UFS_UTILS/pull/118) so that the latest NCEPLIBS can be used for the entire app, as well as Larissa's upcoming PR that will allow us to use the authoritative UFS_UTILS repository. The cmake build is in [this fork](https://github.com/kgerheiser/ufs-srweather-app/tree/feature/cmake), and a PR will be opened once all the UFS_UTILS changes are in place.

**Comment by jwolff-ncar** [comment]
@mkavulich @llpcarson Is there another issue this should be linked to that is being actively worked on? Or does this just need to be updated?

**Comment by mkavulich** [comment]
The top-level cmake build is now in place; the last thing before it is finalized will be this PR for NCEPLIBS which creates files that set up the user's environment for building the app. See that PR for more details: https://github.com/NOAA-EMC/NCEPLIBS/pull/148

**Comment by mkavulich** [comment]
The cmake build is now platform-generic. Users should be able to install NCEPLIBS and point to it with cmake, and this should theoretically work on any machine provided NCEPLIBS is installed successfully and the proper environment variables are set.
